silkworm/db/snapshots/common/bitmask_operators.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <type_traits>
+
+namespace silkworm::snapshots {
```

Comment:
I don't see your point here because including `decompressor.hpp` will add these operands only for `CompressionKind` type, which is what you want in order to be able to specify compound values for it in the user code.
On the other hand, any other type is not affected until you explicitly instantiate the bitmask operators for such type.